OpenEdXMobile/src/test/java/org/edx/mobile/view/WhatsNewFragmentTest.java

```diff
@@ -0,0 +1,77 @@
+package org.edx.mobile.view;
+
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+
+import org.edx.mobile.BuildConfig;
+import org.edx.mobile.R;
+import org.edx.mobile.util.FileUtil;
+import org.edx.mobile.util.Version;
+import org.edx.mobile.whatsnew.WhatsNew;
+import org.edx.mobile.whatsnew.WhatsNewFragment;
+import org.edx.mobile.whatsnew.WhatsNewModel;
+import org.junit.Test;
+import org.robolectric.shadows.support.v4.SupportFragmentTestUtil;
+
+import java.io.IOException;
+import java.lang.reflect.Type;
+import java.text.ParseException;
+import java.util.List;
+
+import static org.assertj.core.api.Java6Assertions.assertThat;
+
+public class WhatsNewFragmentTest extends UiTest {
+    private final String IMAGE_NAME_REGEX = "screen_[0-9]{0,9}";
+
+    @Test
+    public void whatsNewJsonParsingTest() throws IOException, ParseException {
+        final WhatsNewFragment fragment = new WhatsNewFragment();
+        SupportFragmentTestUtil.startVisibleFragment(fragment);
+
+        final String whatsNewJson = FileUtil.loadTextFileFromResources(fragment.getContext(),
+                R.raw.whats_new);
+        final Type type = new TypeToken<List<WhatsNew>>() {}.getType();
+        final List<WhatsNew> list = new Gson().fromJson(whatsNewJson, type);
+
+        // Check all version names in json are valid
+        for(WhatsNew whatsNew : list) {
+            assertThat(new Version(whatsNew.getVersion())).isNotNull();
+        }
+
+        List<WhatsNewModel> messages;
+
+        // What's new content doesn't exist
+        messages = fragment.getWhatsNewMessages("1.10", list);
+        assertThat(messages).isNull();
+
+        // Whats's new content exist
+        messages = fragment.getWhatsNewMessages("2.10.2", list);
+        assertThat(messages).isNotEmpty();
+        validateWhatsNewMessages(messages);
+
+        // If what's new content for current build version exist
+        messages = fragment.getWhatsNewMessages(BuildConfig.VERSION_NAME, list);
+        if (messages != null) {
+            validateWhatsNewMessages(messages);
+        }
+    }
+
+    public void validateWhatsNewMessages(List<WhatsNewModel> messages) {
+        for(WhatsNewModel message : messages) {
+            assertThat(message).isNotNull();
+            assertThat(message.getTitle()).isNotEmpty();
+            assertThat(message.getMessage()).isNotEmpty();
+
+            assertThat(message.getImage()).isNotEmpty();
+            // Check image name is valid
+            assertThat(message.getImage()).containsPattern(IMAGE_NAME_REGEX);
+
+            final List<String> platforms = message.getPlatforms();
+            assertThat(platforms).isNotEmpty();
+            for (String platform : platforms) {
+                // Check platform name is valid (either ios or android)
+                assertThat(platform).isIn("android", "ios");
```

Comment:
The `messages` won't ever have any message besides the ones that have the platform as `android`. This test's dataset should be updated.